Sage/carbon
# refactor(button-toggle-group): convert to typescript

### Proposed behaviour

- `ButtonToggleGroup`
   - Convert component to typescript
   - Remove old documentation files from component directory
   - Define type declaration `.d.ts` file for `RadioButtonMapper`, which is used by `ButtonToggleGroup` internally
   - Rename `ButtonToggleGroupStyle` to `StyledButtonToggleGroup` for consistency
- `ButtonToggle` 
   - Modify stories where multiple buttons are used - to match our recommended structure of only wrapping them with `ButtonToggleGroup`
   - Make `children` an optional prop and add invariant message to `ButtonToggle` for childless and iconless buttons - so a missing children error is not thrown for [icon-only buttons](https://carbon.sage.com/?path=/story/button-toggle--icon-only).
   - Add disclaimer to icon-only story that icon-only buttons are not a recommended pattern for accessibility reasons.

### Current behaviour

- `ButtonToggleGroup`
   - Is written in javascript
   - Directory contains old documentation which is not used by storybook  

### Checklist




#### QA


### Testing instructions

- Check no regressions have been introduced as a result of these changes
- Check error message of missing children has been resolved using the following codesandbox: https://codesandbox.io/s/keen-rain-nfmlct?file=/src/index.tsx
